src/portal/src/app/base/left-side-nav/projects/create-project/create-project.component.ts

```diff
@@ -297,7 +312,39 @@ export class CreateProjectComponent
         }
     }
 
+    // **Added method for validating bandwidth input**
+    validateBandwidth(): void {
+        const value = Number(this.speedLimit);
+        if (
+            isNaN(value) ||
+            (!Number.isInteger(value) && value !== -1) ||
+            (value <= 0 && value !== -1)
+        ) {
+            this.bandwidthError =
+                'Please enter -1 or an integer  greater than 0.';
+        } else {
+            this.bandwidthError = null;
+        }
+    }
+    convertSpeedValue(realSpeed: number): number {
+        if (this.speedLimitUnit == BandwidthUnit.MB) {
```

Comment:
Did you verify when you select MBs, the value in payload is not converted?
It should be `this.selectedSpeedLimitUnit`, not `this.speedLimitUnit`